obspy/io/nordic/core.py

```diff
@@ -0,0 +1,1379 @@
+                    for i in spec_inf]
+        stachan = str('.'.join([spec_str[4:9].strip(),
+                               ''.join(spec_str[9:13].split())]))
+        if stachan in stachans:
+            info = [i for i in spec_inf if
+                    str(i['station']) == str(spec_str[4:9].strip()) and
+                    str(i['channel']) == str(''.join(spec_str[9:13].
+                                                     split()))][0]
+            new_stachan = False
+        else:
+            info = {}
+            info['station'] = spec_str[4:9].strip()
+            info['channel'] = ''.join(spec_str[9:13].split())
+            new_stachan = True
+        if spec_str[14] == 'T':
+            info['starttime'] = origin_date + (int(spec_str[15:17]) * 3600) +\
+                (int(spec_str[17:19]) * 60) + int(spec_str[19:21])
+            if info['starttime'] < event.origins[0].time:
+                # Wrong day, case of origin at end of day
+                info['starttime'] += 86400
+            info['kappa'] = _float_conv(spec_str[23:30])
+            info['distance'] = _float_conv(spec_str[32:38])
+            if spec_str[38:40] == 'VS':
+                info['velocity'] = _float_conv(spec_str[40:46])
+                info['velocity_type'] = 'S'
+            elif spec_str[38:40] == 'VP':
+                info['velocity'] = _float_conv(spec_str[40:46])
+                info['velocity_type'] = 'P'
+            info['density'] = _float_conv(spec_str[48:54])
+            info['Q0'] = _float_conv(spec_str[56:62])
+            info['QA'] = _float_conv(spec_str[64:70])
+        elif spec_str[14] == 'M':
+            info['moment'] = _float_conv(spec_str[16:22])
+            if info['moment'] == 999:
+                info['moment'] = np.nan
+            else:
+                info['moment'] = 10 ** info['moment']
+            info['stress_drop'] = _float_conv(spec_str[24:30])
+            info['spectral_level'] = _float_conv(spec_str[32:38])
+            if info['spectral_level'] == 999:
+                info['spectral_level'] = np.nan
+            else:
+                info['spectral_level'] = 10 ** info['spectral_level']
+            info['corner_freq'] = _float_conv(spec_str[40:46])
+            info['source_radius'] = _float_conv(spec_str[47:54])
+            info['decay'] = _float_conv(spec_str[56:62])
+            info['window_length'] = _float_conv(spec_str[64:70])
+            info['moment_mag'] = _float_conv(spec_str[72:78])
+        if new_stachan:
+            spec_inf.append(info)
+    return spec_inf
+
+
+def read_nordic(select_file):
+    """
+    Read a catalog of events from a Nordic formatted select file.
+    Generates a series of temporary files for each event in the select file.
+
+    :type select_file: str
+    :param select_file: Nordic formatted select.out file to open
+
+    :return: catalog of events
+    :rtype: obspy.core.event.Catalog
+    """
+    from tempfile import NamedTemporaryFile
```

Comment:
Please try not to use actual temporary files. Consider refactoring the logic to for example use `io.BytesIO` (should be very simple) or maybe just pass the raw file pointer to the underlying functions?